web/src/components/ui/scroll-to-top.tsx

```diff
@@ -0,0 +1,59 @@
+"use client";
+
+import Link from "next/link";
+import { ReactNode, useCallback } from "react";
+
+import { css } from "@/styled-system/css";
+import { scrollToTop } from "@/utils/scroll";
+
+interface ScrollToTopProps {
+  children?: ReactNode;
+}
+
+export function ScrollToTop({ 
+  children = "scroll to top"
+}: ScrollToTopProps) {
+  const handleClick = useCallback((e: React.MouseEvent<HTMLAnchorElement>) => {
+    e.preventDefault();
+    scrollToTop();
+  }, []);
```

Comment:
I really like this progressively-enhanced approach with the href as # for non JS users, and the smooth scroll for everyone else!